File: 590. N-ary Tree Postorder Traversal/Solution590.java
import java.util.ArrayList;
import java.util.List;

class Node {
    public int val;
    public List<Node> children;

    public Node() {
    }

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};

class Solution590 {

    public static List<Integer> postorder(Node root) {
        ArrayList<Integer> list = new ArrayList<>();
        if (root == null) {
            return list;
        }

        dfs(root, list);

        return list;

    }

    private static void dfs(Node root, ArrayList<Integer> list) {

        for (Node node : root.children) {
            dfs(node, list);
        }
        list.add(root.val);

    }

    public static void main(String[] args) {

    }
}

File: 590. N-ary Tree Postorder Traversal/Solution590.py
class Node(object):
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children


class Solution(object):
    def postorder(self, root):
        
        if not root:
            return []
        
        ans = []

        def dfs(root):

            for node in root.children:
                dfs(node)

            ans.append(root.val)

        dfs(root)

        return ans
        
            
        